src/main/groovy/actions/TransferActions.groovy

```diff
@@ -112,6 +157,89 @@ class TransferActions {
         LabwareActions.updateLabware(destinationLabware)
     }
 
+    private static createTransferMapForCombine(numberOfSourcePlates,
+                                            sourceLayout, plateNumber) {
+        def numberOfRows = sourceLayout.row
+        def numberOfColumns = sourceLayout.column
+
+        def transferMap = [];
+        (1..numberOfRows).each { row ->
+            def transferMapRow =
+                (1..numberOfColumns).collect { column ->
+                    def sourceRow = (char) (64 + row)
+                    def sourceColumn = column
+
+                    def destinationRow
+                    def destinationColumn
+                    switch (plateNumber) {
+                        case 0:
```

Comment:
Rather than these cases you could do plateNumber div and mod 2 to decide whether to subtract 1.